internal/handlers/common/aggregations/operators/accumulators/sum.go

```diff
@@ -77,7 +95,18 @@ func (s *sum) Accumulate(iter types.DocumentsIterator) (any, error) {
 			return nil, lazyerrors.Error(err)
 		}
 
-		if s.expression != nil {
+		switch {
+		case s.operator != nil:
+			v, err := s.operator.Process(doc)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
Let's cover this error case in a test!

Reply:
This is unreachable as the Process is called before this line to validate everything. I believe we should refactor that in the next issue and move all of the validations in operators into constructor or separate function though